src/Plugin/BaseUriPlugin.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Http\Client\Common\Plugin;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * This plugin combines the host and path plugins.
+ *
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class BaseUriPlugin implements Plugin
+{
+    /**
+     * @var AddHostPlugin
+     */
+    private $addHostPlugin;
+
+    /**
+     * @var AddPathPlugin
+     */
+    private $addPathPlugin = null;
+
+    /**
+     * @param UriInterface $host
+     * @param array        $hostConfig Config for AddHostPlugin. @see AddHostPlugin::configureOptions
+     */
+    public function __construct(UriInterface $host, array $hostConfig = [])
+    {
+        $this->addHostPlugin = new AddHostPlugin($host, $hostConfig);
+
+        if ($host->getPath()) {
```

Comment:
we should rtrim($path, '/') here. or throw an exception on http://domain.com/ (it makes no sense to prepend just / to a uri)